x-pack/plugins/remote_clusters/public/store/actions/remove_clusters.js

```diff
@@ -27,33 +25,46 @@ export const removeClusters = (names) => async (dispatch, getState) => {
     type: REMOVE_CLUSTERS_START,
   });
 
-  try {
-    await Promise.all([
-      sendRemoveClustersRequest(names),
-      // Wait at least half a second to avoid a weird flicker of the saving feedback.
-      new Promise(resolve => setTimeout(resolve, 500)),
-    ]);
-  } catch (error) {
-    dispatch({
-      type: REMOVE_CLUSTERS_FAILURE,
-      payload: { error }
-    });
+  const removalSuccesses = [];
+  const removalErrors = [];
+  const removeClusterRequests = names.map(name => {
+    sendRemoveClusterRequest(name)
+      .then(() => removalSuccesses.push(name))
+      .catch(() => removalErrors.push(name));
+  });
+
+  await Promise.all([
```

Comment:
This would be better if done server side like I did with auto-follow pattern. We can maybe leave it as an optimization fix for later though.